spinoso-time/src/time/tzrs/math.rs

```diff
@@ -0,0 +1,491 @@
+    }
+}
+
+impl Add<i8> for Time {
+    type Output = Self;
+
+    fn add(self, seconds: i8) -> Self::Output {
+        self + i64::from(seconds)
+    }
+}
+
+impl Add<u8> for Time {
+    type Output = Self;
+
+    fn add(self, seconds: u8) -> Self::Output {
+        self + u64::from(seconds)
+    }
+}
+
+impl Add<i16> for Time {
+    type Output = Self;
+
+    fn add(self, seconds: i16) -> Self::Output {
+        self + i64::from(seconds)
+    }
+}
+
+impl Add<u16> for Time {
+    type Output = Self;
+
+    fn add(self, seconds: u16) -> Self::Output {
+        self + u64::from(seconds)
+    }
+}
+
+impl Add<i32> for Time {
+    type Output = Self;
+
+    fn add(self, seconds: i32) -> Self::Output {
+        self + i64::from(seconds)
+    }
+}
+
+impl Add<u32> for Time {
+    type Output = Self;
+
+    fn add(self, seconds: u32) -> Self::Output {
+        self + u64::from(seconds)
+    }
+}
+
+impl Add<i64> for Time {
+    type Output = Self;
+
+    fn add(self, seconds: i64) -> Self::Output {
+        if seconds.is_negative() {
+            let seconds = seconds
+                .checked_neg()
+                .and_then(|secs| u64::try_from(secs).ok())
+                .expect("Duration too large");
+            self - Duration::from_secs(seconds)
+        } else {
+            let seconds = u64::try_from(seconds).expect("Duration too large");
+            self + Duration::from_secs(seconds)
+        }
+    }
+}
+
+impl Add<u64> for Time {
+    type Output = Self;
+
+    fn add(self, seconds: u64) -> Self::Output {
+        let duration = Duration::from_secs(seconds);
+        self + duration
+    }
+}
+
+impl Add<f32> for Time {
+    type Output = Self;
+
+    fn add(self, seconds: f32) -> Self::Output {
+        self + f64::from(seconds)
+    }
+}
+
+impl Add<f64> for Time {
+    type Output = Self;
+
+    fn add(self, seconds: f64) -> Self::Output {
+        if seconds.is_sign_positive() {
+            self + Duration::from_secs_f64(seconds)
+        } else {
+            let seconds = -seconds;
+            self - Duration::from_secs_f64(seconds)
+        }
+    }
+}
+
+impl Sub<Duration> for Time {
+    type Output = Self;
+
+    fn sub(self, duration: Duration) -> Self::Output {
+        let unix_time = self.inner.unix_time();
+        let nanoseconds = self.inner.nanoseconds();
+        let offset = self.offset;
+
+        let duration_seconds = i64::try_from(duration.as_secs()).expect("Duration too large");
+        let duration_subsecs = duration.subsec_nanos();
+
+        let mut seconds = unix_time.checked_sub(duration_seconds).expect("Duration too large");
+        let nanoseconds = if let Some(nanos) = nanoseconds.checked_sub(duration_subsecs) {
+            nanos
+        } else {
+            seconds -= 1;
+            nanoseconds + NANOS_IN_SECOND - duration_subsecs
+        };
+
+        Self::Output::with_timespec_and_offset(seconds, nanoseconds, offset)
```

Comment:
Now that I think about it, maybe we keep these `ops` impls but have them delegate to a fallible method on `Time` and only unwrap here?

Reply:
Yeah, I was also struggling with the overflow aspect of these ops and how to integrate them. I'll throw it into the errors PR (I've updated it as a TODO)